test/org/zaproxy/zap/extension/pscanrulesAlpha/UsernameIdorScannerUnitTest.java

```diff
@@ -0,0 +1,120 @@
+ *
+ * Copyright 2017 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.pscanrulesAlpha;
+
+import static org.mockito.Mockito.mock;
+
+import static org.junit.Assert.assertEquals;
+
+import org.apache.commons.httpclient.URI;
+import org.apache.commons.httpclient.URIException;
+import org.junit.Before;
+import org.junit.Test;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+import org.zaproxy.zap.extension.users.ExtensionUserManagement;
+
+public class UsernameIdorScannerUnitTest extends PassiveScannerTest {
+
+	private ExtensionUserManagement extensionUserMgmtMock;
+	private HttpMessage msg;
+
+	// Hashes in lower case for "guest" without quotes
+	private static final String GUEST_MD5 = "084e0343a0486ff05530df6c705c8bb4";
+	private static final String GUEST_SHA1 = "84983c60f7daadc1cb8698621f802c0d9f9a3c3c295c810748fb048115c186ec";
+
+	@Before
+	public void before() throws URIException {
+
+		extensionUserMgmtMock = mock(ExtensionUserManagement.class);
+
+		((UsernameIdorScanner) rule).setExtensionUserManagement(extensionUserMgmtMock);
+		((UsernameIdorScanner) rule).setUsers("guest");
+
+		HttpRequestHeader requestHeader = new HttpRequestHeader();
+		requestHeader.setURI(new URI("http://example.com", false));
+
+		msg = new HttpMessage();
+		msg.setRequestHeader(requestHeader);
+	}
+
+	@Override
+	protected PluginPassiveScanner createScanner() {
+		return new UsernameIdorScanner();
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseHasNoRelevantContent() {
+		// Given
+		msg.setResponseBody("Some text <h1>Some Title Element</h1>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertEquals(alertsRaised.size(), 0);
+	}
+
+	@Test
+	public void shouldNotRaiseAlertIfResponseContainsIrrelevantHash() {
+		// Given - "Guest" with a leading cap
+		msg.setResponseBody("Some text <h1>Some Title Element</h1><i>adb831a7fdd83dd1e2a309ce7591dff8</i>");
+		// When
+		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
+		// Then
+		assertEquals(alertsRaised.size(), 0);
+	}
+
+	@Test
+	public void shouldRaiseAlertIfResponseContainsRelevantMd5Hash() {
+		// Given
+		msg.setResponseBody("Some text <h1>Some Title Element</h1><i>084E0343A0486FF05530DF6C705C8BB4</i>");
```

Comment:
So interestingly the Pattern in the scanner will match if all alpha chars are lower or upper, but it won't if mixed?(i.e.: if I make one of the Bs toward the end lower case it doesn't match.) I doubt mixed will be encountered "in the wild" but it would be nice if that worked too. Any insight?